defenz_alerts/main.py

```diff
@@ -0,0 +1,491 @@
+        default=CONFIG['DEFENZ']['CLIENT_ID'],
+        help='Defenz client id (customer name). Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-s',
+        '--client-secret',
+        default=CONFIG['DEFENZ']['CLIENT_SECRET'],
+        help='Defenz client secret. Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-e',
+        '--report-email',
+        default=CONFIG['DEFENZ']['REPORT_EMAIL'],
+        help='The report will be sent to this email address '
+             'if network report email is not set. '
+             'Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-n',
+        '--network-ids',
+        nargs='*',
+        help='List of network ids. Default all networks'
+    )
+
+    args_parser.add_argument(
+        '-r',
+        '--report-types',
+        nargs='*',
+        default=CONFIG['DEFENZ']['REPORT_TYPES'].split(',')
+        if CONFIG['DEFENZ']['REPORT_TYPES']
+        else [MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS],
+        choices=[MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS, WEB_FILTER_BLOCKS],
+        help='List of report types. Default {} and {}'.format(
+            MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS)
+    )
+
+    args_parser.add_argument(
+        '-i',
+        '--interval',
+        default=CONFIG['DEFENZ']['INTERVAL'],
+        help='Report interval in minute. Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-v',
+        '--verbose',
+        default=False,
+        action='store_true',
+        help='Write the logs in console as well. Otherwise the logs only will '
+             'be written in the log files. The log files are located in '
+             '$HOME/defenz'
+    )
+
+    return args_parser.parse_args()
+
+
+def main():
+    """
+    The main function of the script.
+    """
+
+    try:
+        args = get_command_line_arguments()
+
+        if args.verbose:
+            console_handler = logging.StreamHandler()
+            console_handler.setLevel(logging.DEBUG)
+            console_handler.setFormatter(
+                logging.Formatter('%(asctime)s - %(levelname)s: %(message)s'))
+            LOGGER.addHandler(console_handler)
+
+        validate_arguments(args)
+
+        AUTHENTICATOR.login(
+            args.username, args.password,
+            args.client_id, args.client_secret
+        )
+
+        networks = []
+        if args.network_ids:
+            for network_id in args.network_ids:
+                networks.append(get_network(network_id))
+        else:
+            networks = get_networks()
+
+        # We need to be consistent about report time throughout the script
+        now = datetime.now()
+
+        for net in networks:
+            try:
+                reports = get_all_reports(net['id'], now, args.interval)
+                for report_type in args.report_types:
+                    if reports[report_type]['results']:
```

Comment:
Not a big thing, but I wonder if we could be cautious here and trim the report_type from the report_type in case someone put spaces between the commas in config such as `REPORT_TYPES = MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS` or `REPORT_TYPES = MALWARE_PHISHING_BLOCKS , BOT_NET_BLOCKS`